### PR TITLE
[TypeDeclaration] Use return bool for ParentClassMethodTypeOverrideGuard::hasParentClassMethod(), return ?MethodReflection for ParentClassMethodTypeOverrideGuard::getParentClassMethod()

### PR DESCRIPTION
**Pull request by samsonasik**
Ref https://github.com/rectorphp/rector-src/pull/4619#pullrequestreview-1565014006

### CODE REVIEW

File: packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php [hunk 1]
```diff
@@ -27,15 +26,15 @@ public function __construct(
...
+    public function hasParentClassMethod(ClassMethod $classMethod): bool
     {
         try {
             $parentClassMethod = $this->resolveParentClassMethod($classMethod);
 
             return $parentClassMethod instanceof MethodReflection;
         } catch (UnresolvableClassException) {
-            // we don't know all involved parents.
-            return null;
+            // we don't know all involved parents, marking as parent exists
```
Comment (staabm) [inline packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php]:
```suggestion
            // we don't know all involved parents,
            // marking as parent exists which usually means the method is guarded against overrides.
```

File: src/DependencyInjection/LazyContainerFactory.php [hunk 2]
```diff
@@ -604,9 +603,10 @@ private function registerTagged(Container $container, array $classes, string $ta
         }
     }
 
-    private function createPHPStanServices(LazyRectorConfig $container): void
+    private function createPHPStanServices(LazyRectorConfig $lazyRectorConfig): void
     {
-        $container->singleton(
+        $container = null;
+        $lazyRectorConfig->singleton(
```
Comment (staabm) [inline src/DependencyInjection/LazyContainerFactory.php]: looks like a bug was introduced here

**Review by staabm — CHANGES_REQUESTED** [review]

File: src/DependencyInjection/LazyContainerFactory.php (hunk 2 above)
Comment (samsonasik) [reply src/DependencyInjection/LazyContainerFactory.php]: Yes, I tested `ReturnTypeFromStrictParamRector` to use `ClassMethodReturnTypeOverrideGuard` instead as it change return type https://github.com/rectorphp/rector-src/pull/4703/files#diff-a5951960805d541e1e180ef2e37996275c1d53f8f6d3186435fce3b14b1727d6

Comment (samsonasik) [reply src/DependencyInjection/LazyContainerFactory.php]: @staabm the bug is happen on main branch as well ..., I am looking into it ;)

Comment (samsonasik) [reply src/DependencyInjection/LazyContainerFactory.php]: @TomasVotruba it seems it related to @TomasVotruba change on PR https://github.com/rectorphp/rector-src/actions/runs/5784912801/job/15676537339

Comment (TomasVotruba) [reply src/DependencyInjection/LazyContainerFactory.php]: It crashes when variable is renamed and deafult null is added at the same time. It collision of 2 rules, but we can skip this manually now.

Comment (samsonasik) [reply src/DependencyInjection/LazyContainerFactory.php]: @TomasVotruba ok, I skipped it https://github.com/rectorphp/rector-src/pull/4703/commits/e606b275178957d74cfcb3e8529013589dccae7c

**Pull request merged**